daemon/images/images.go

```diff
@@ -205,19 +201,31 @@ func (i *ImageService) Images(imageFilters filters.Args, all bool, withExtraAttr
 			// NOTE: By default, Containers is -1, or "not set"
 			summary.Containers = containers
 
-			// count layer references
+			summaryMap[img] = summary
+		}
+		summaries = append(summaries, summary)
+	}
+
+	if withExtraAttrs {
+		allLayers := i.layerStore.Map()
+		layerRefs := make(map[layer.ChainID]int, len(allLayers))
+
+		allImages := selectedImages
+		if danglingOnly {
+			// selectedImages include only dangling images, but
+			// we need to consider all existing images to correctly perform reference counting.
+			allImages = i.imageStore.Map()
+		}
+		// Count layer references across all known images
+		for _, img := range allImages {
```

Comment:
shouldn't this be just `range i.imageStore.Map()` and no `danglingOnly` check in lines above

Reply:
I guess you want to avoid calling `Map()` again for the non-dangling case. Maybe add a comment for it.

Reply:
Updated the comment